src/realm/sync/client.cpp

```diff
@@ -960,22 +960,19 @@ void SessionWrapper::refresh(std::string signed_access_token)
     // Thread safety required
     REALM_ASSERT(m_initiated);
 
-    util::bind_ptr<SessionWrapper> self{this};
-    auto handler = [self = std::move(self), token = std::move(signed_access_token)] {
+    m_client.get_service().post([self = util::bind_ptr(this), token = std::move(signed_access_token)] {
         REALM_ASSERT(self->m_actualized);
         if (REALM_UNLIKELY(!self->m_sess))
             return; // Already finalized
         self->m_signed_access_token = std::move(token);
         SessionImpl& sess = *self->m_sess;
         ClientImpl::Connection& conn = sess.get_connection();
         // FIXME: This only makes sense when each session uses a separate connection.
-        conn.update_connect_info(self->m_http_request_path_prefix, self->m_virt_path,
```

Comment:
The actual change was removing `self->m_virt_path` argument, but while I was here, I also moved the lambda into the post() call